USER-INTERFACE-INTERFACE.md

```diff
@@ -640,41 +670,73 @@ If the UI is remembering the database password, it should forget it when this me
 ```
 "payload": {
     "dbPassword": <string>,
-    "mnemonicPhrase": [
-        <string>,
-        <string>,
-        [...]
-    ],
-    "mnemonicPassphraseOpt": <optional string>,
-    "mnemonicPhraseLanguage": <string>,
-    "consumingDerivationPath": <string>,
-    "earningWallet": <string>
+    "seedSpecOpt": {
+        "mnemonicPhrase": [
+            <string>,
+            <string>,
+            [...]
+        ],
+        "mnemonicPhraseLanguage": <string>,
+        "mnemonicPassphraseOpt": <optional string>
+    },
+    "consumingDerivationPathOpt": <optional string>,
+    "consumingPrivateKeyOpt": <optional string>,
+    "earningDerivationPathOpt": <optional string>,
+    "earningAddressOpt": <optional string>,
 }
 ```
 ##### Description:
 This message directs the Node to set its wallet pair to a preexisting pair of wallets described in the message.
 If the database already contains a wallet pair, the wallet recovery will fail.
 
+Each wallet can be recovered in one of two ways.
+
+The consuming wallet can be recovered by specifying the mnemonic phrase of its seed and its derivation path, or by
+giving its private key. (The Node needs the private key of the consuming wallet so that it can withdraw funds from it
+to pay bills.)
+
+The earning wallet can be recovered by specifying the mnemonic phrase of its seed and its derivation path, or by
+giving its address. (The Node does not need (and will not accept) the private key of the earning wallet, because all
+it ever has to do is deposit funds in it from other Nodes. The private key can be derived from the seed and the
+derivation path, but the Node does not store it.)
+
+Private key/address are preferred; if you provide both that and the derivation path for either wallet, the derivation
+path will be ignored.
+
+If you expect the derivation path to be used for either wallet, you must also provide the seed specification.
+
 `dbPassword` is the current database password. If this is incorrect, the wallet recovery will fail.
 
-`mnemonicPhrase` is the mnemonic phrase that was used to generate the consuming wallet and possibly the earning
-wallet as well. It must have 12, 15, 18, 21, or 24 words.
+`seedSpecOpt` gives the parameters for generating the seed. This only makes sense if one or more of the
+derivation paths is supplied. If no derivation paths are supplied, this parameter is ignored.
+
+`mnemonicPhrase` is the mnemonic phrase that represents the seed. It must have 12, 15, 18, 21, or 24 words.
+
+`mnemonicPhraseLanguage` is the language in which the mnemonic phrase is supplied. Acceptable values are
```

Comment:
this should not be a required value under seedSpecOpt parameters, (as should be defaulted to English), but its worth making sure this is the case @dnwiebe ?